tests/e2e/spec/tonedear.e2e.spec.js

```diff
@@ -0,0 +1,233 @@
+/* eslint-disable no-undef */
+import { By, until } from 'selenium-webdriver';
+import assert from 'assert';
+import paths from '../paths.js';
+
+const BROWSERSTACK = !!process.env.BROWSERSTACK_LOCAL_IDENTIFIER;
+const port = BROWSERSTACK ? '8099' : '8080';
+const tonedearBaseFile = BROWSERSTACK ? '/tests/zims/tonedear/tonedear.com_en_2024-09.zim' : paths.tonedearBaseFile;
+
+/**
+ * Run the tests
+ * @param {WebDriver} driver Selenium WebDriver object
+ * @param {Array} modes Array of modes to run the tests in
+ * @param {boolean} keepDriver Whether to keep the driver open after the tests have run
+ * @returns {Promise<void>} A Promise for the completion of the tests
+ */
+function runTests (driver, modes, keepDriver) {
+    let browserName, browserVersion;
+    driver.getCapabilities().then(function (caps) {
+        browserName = caps.get('browserName');
+        browserVersion = caps.get('browserVersion');
+        console.log('\nRunning Tonedear tests on: ' + browserName + ' ' + browserVersion);
+    });
+
+    // Set the implicit wait to 3 seconds
+    driver.manage().setTimeouts({ implicit: 3000 });
+
+    // Run in both jquery and serviceworker modes by default
+    if (!modes) {
+        modes = ['jquery', 'serviceworker'];
+    }
+
+    modes.forEach(function (mode) {
+        // eslint-disable-next-line prefer-const
+        let serviceWorkerAPI = true;
```

Comment:
@THEBOSS0369 I'm sorry, you've misunderstood me! I wasn't asking you merely to convert this from const to let. I was saying that you actually need to test whether the serviceWorkerAPI is available and then set it to true or false accordingly, before running the rest of the tests!

For an example of how this is done, see in the Gutenberg spec here: https://github.com/kiwix/kiwix-js/blob/main/tests/e2e/spec/gutenberg_ro.e2e.spec.js#L131